### DISCUSSION TITLE
Add pagination of subcategories, on the category page

### DISCUSSION THREAD

**Discussion by mtheinmix** [PrestaShop/PrestaShop#40160] [Ideas]
If there are many subcategories on the category page (1000+), this loads the page heavily

Suggestion adding pagination of subcategories as done with products

For example, make it possible to select the number of displayed subcategories on the category page.
For example: 50 subcategories on 1 page, 50 more on the second page, etc. (Pagination of subcategories as with products, only with subcategories)

Default, all subcategories are displayed at once on one page, regardless of their number; this is problematic when there are a large number of subcategories.

[a comment by an automated account is not shown]

**Comment by Sinepel** [comment]
Hello!

This is a very special case, I don't know if it would be useful to implement this within Prestashop directly.

Ping @PrestaShop/prestashop-product-team /  @LouiseBonnard / @marionf , wdyt ?

**Comment by khouloudbelguith** [comment]
Hi @Sinepel,

Thanks for your report.
Ping @PrestaShop/prestashop-product-team what do you think?

Thanks!

**Comment by mlemanczyk** [comment]
This is very much needed in the admin. In our store it takes anywhere up to **30 minutes(!) or more** to load the product page. As you can imagine, it's a real challenge to work with the products using PrestaShop admin. We have 254+ categories.

You may also want to check loading of product feature values. Some of our features has as many values as many products we have. And we have 40k+ products. When feature values are loaded all together on the product page, it may take time...

I believe all of that contributes to extremely slow product page loading.

**Comment by LouiseBonnard** [comment]
Hi @mtheinmix, thanks for the feedback! Here are some interesting figures about how PrestaShop reacts to different numbers of categories: https://github.com/PrestaShop/PrestaShop/issues/19281#issuecomment-633550499. 

Performance issues might unfortunately happen when there is a lot of items to load or requests to send. Have you experienced the pagination of subcategories on your end yet? If yes, have you noticed a significant performance improvement in addition to a better user experience?

**Comment by mtheinmix** [comment]
> Hi @mtheinmix, thanks for the feedback! Here are some interesting figures about how PrestaShop reacts to different numbers of categories: [#19281 (comment)](https://github.com/PrestaShop/PrestaShop/issues/19281#issuecomment-633550499).
> 
> Performance issues might unfortunately happen when there is a lot of items to load or requests to send. Have you experienced the pagination of subcategories on your end yet? If yes, have you noticed a significant performance improvement in addition to a better user experience?

Hello, no, I have not tested it, I just suggested an idea)
I myself tried to implement this functionality based on the pagination guides in other sections, but, unfortunately, nothing came of it, I do not have enough knowledge in programming..

**Comment by matks** [comment]
> This is very much needed in the admin. In our store it takes anywhere up to **30 minutes(!) or more** to load the product page. As you can imagine, it's a real challenge to work with the products using PrestaShop admin. We have 254+ categories.
> 
> You may also want to check loading of product feature values. Some of our features has as many values as many products we have. And we have 40k+ products. When feature values are loaded all together on the product page, it may take time...
> 
> I believe all of that contributes to extremely slow product page loading.

Err ... 😅 we are talking about categories listing here, not product page
